src/main/kotlin/frc/robot/subsystems/ReefCommands.kt

```diff
@@ -90,7 +99,12 @@ fun l3(outtakeTrigger: Trigger): Command =
     parallel(elevator.l3(), wrist.l3()).andThen(scoreCoral(outtakeTrigger))
 
 fun l4(outtakeTrigger: Trigger): Command =
-    parallel(elevator.l4(), wrist.l4()).andThen(scoreCoralL4(outtakeTrigger))
+    parallel(elevator.l4(), wrist.l4(), extender.l4())
+        .until(extender.atSetpoint)
+        .andThen(WaitCommand(0.5))
+        .until(extender.hasBeenPushed)
+        .andThen(DriveCommands.lockWithX(swerveDrive))
+        .andThen(scoreCoralL4(outtakeTrigger))
```

Comment:
About `parallel(elevator.l4(), wrist.l4(), extender.l4()).until(extender.atSetpoint)`, perhaps you meant to use `waitUntil()` there? Because all the commands in the parallel group are run once so `until()` has no effect here.
Also, using `sequence()` is cleaner here, and `waitSeconds()` is a bit clearer:
```suggestion
    sequence(
        parallel(elevator.l4(), wrist.l4(), extender.l4()),
        waitUntil(extender.atSetpoint),
        waitSeconds(0.5),
        waitUntil(extender.hasBeenPushed),
        DriveCommands.lockWithX(swerveDrive),
        scoreCoralL4(outtakeTrigger)
    )
```